### PR TITLE
How We Use Node.js for Backend Development

### PR DESCRIPTION
**Pull request by Snailed**

### CODE REVIEW

File: how-we-use-nodejs.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+- Migrations are only to be generated using the TypeORM CLI utility: This ensures no discrepancies between the schema created by the migrations compared to the one defined by the entities.
+- Migrations are run automatically on server startup after every deployment.
+- Migrations are small and preferably only affects 1 table at a time.
+- Migrations should always be reversable to ensure no breaking changes.
+
+## PostgreSQL
+We use managed SQL servers through Google Cloud SQL. We avoid [common mistakes](https://wiki.postgresql.org/wiki/Don't_Do_This), although our ORM mostly saves us from these.
+
+## Terraform
+We use Terraform for managing all of our infrastructure declaratively, and run our changes on every push to staging or production. We store the Terraform state in the `gs://terraform-state-kvalifik` bucket in the "Terraform Remote State Storage" Google Cloud Project. For a project that uses Terraform well, check [Testamentegenerator](https://github.com/Kvalifik/testamentegenerator-backend). We seperate our environments using `terraform workspace`. All terraform configurations should be able to easily be run in multiple environments.
```
Comment (nikolajvonholck) [inline how-we-use-nodejs.md]: We don't use Terraform workspaces. We have a different google cloud project for each environment.

File: how-we-use-nodejs.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+### Test files
+Each end-2-end test file should be stored in the `test` folder and cover one controller. It should be named `[controller-name].e2e-spec.ts`. Each unit test file should be stored in the same folder as the unit and be called `[unit-name].spec.ts`.
+We create mocks for many of our services. Each mock file should be called `[service-name].service.mock.ts` and be stored alongside the service.
+
+### Other
+Guards, pipes, controllers and modules all use the naming convention `[name].[type].ts`
+
+## Docker
+We use Docker for containerizing our run-time environment. We base our images on the [node:alpine](https://hub.docker.com/_/node) images and always use the latest LTS version. We build our docker images on every push to a branch in our CD flow.
+We use `docker-compose` to manage our docker containers during development.
```
Comment (nikolajvonholck) [inline how-we-use-nodejs.md:43]: You can use `docker compose` (without `-`)

**Review by nikolajvonholck — CHANGES_REQUESTED** [review]
See my comments.

File: how-we-use-nodejs.md (hunk 2 above)
Comment (Snailed) [reply how-we-use-nodejs.md:43]: not per Docker version 20.10.7 on Arch Linux

File: how-we-use-nodejs.md (hunk 1 above)
Comment (Snailed) [reply how-we-use-nodejs.md]: I got terraform workspaces to work with CI/CD on Kvalifik Data Warehouse, and I'm also getting it to work on the "How to setup a backend using Terraform" page

Comment (nikolajvonholck) [reply how-we-use-nodejs.md]: Let's not use workspaces for now, as discussed.

**Review by nikolajvonholck — APPROVED** [review]

**Pull request merged**